staging/src/k8s.io/apimachinery/pkg/api/validation/objectmeta.go

```diff
@@ -261,6 +289,28 @@ func ValidateObjectMetaAccessorUpdate(newMeta, oldMeta metav1.Object, fldPath *f
 		allErrs = append(allErrs, field.Invalid(fldPath.Child("deletionTimestamp"), newMeta.GetDeletionTimestamp(), "field is immutable; may only be changed via deletion"))
 	}
 
+	oldInit := oldMeta.GetInitializers()
+	newInit := newMeta.GetInitializers()
+	switch {
+	case oldInit == nil && newInit != nil:
+		// Initializers may not be set on new objects
+		allErrs = append(allErrs, field.Invalid(fldPath.Child("initializers"), nil, "field is immutable once all initialization has completed"))
```

Comment:
The admission controller halts before other admission controllers can run, AND before validation is run.  I don't think we should allow other admission controllers to observe a completely invalid transition.

We discussed this a while ago, but we should probably run validation prior to beginning admission.  At that point, this would then be the only code needed.  Much bigger change.